runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractQueuedStreamView.java

```diff
@@ -231,8 +182,324 @@ public long append(Object object,
+            updatePointer(entries.get(entries.size() - 1));
+            queueAddressManager.validateGlobalPointerPosition(getCurrentGlobalPosition());
+        }
+
+        // And return the entries.
+        return entries;
+    }
+
+    /** Update the global pointer, given an entry.
+     *
+     * @param data  The entry to use to update the pointer.
+     */
+    private void updatePointer(final ILogData data) {
+        // Update the global pointer, if it is non-checkpoint data.
+        if (data.getType() == DataType.DATA && !data.hasCheckpointMetadata()) {
+            // Note: here we only set the global pointer and do not validate its position with respect to the trim mark,
+            // as the pointer is expected to be moving step by step (for instance when syncing a stream up to maxGlobal)
+            // The validation is deferred to these methods which call it in advance based on the expected final position
+            // of the pointer.
+            queueAddressManager.setGlobalPointer(data.getGlobalAddress());
+        }
+    }
+
+    /** Retrieve the next entries in the stream.
+     *
+     * <p>This function is designed to read all entries in the read queue
+     *    in parallel.
+     *
+     * <p>The default implementation simply calls getNextEntry.
+     *
+     * @param maxGlobal         The maximum global address to read to.
+     *
+     * @return                  A list of the next entries for this context
+     */
+    protected List<ILogData> getNextEntries(long maxGlobal) {
+        // Scan backward in the stream to find interesting
+        // log records less than or equal to maxGlobal.
+        // Boolean includes both CHECKPOINT & DATA entries.
+        boolean readQueueIsEmpty = !fillReadQueue(maxGlobal);
+
+        // If maxGlobal is before the checkpoint position, throw a
+        // trimmed exception
+        if (maxGlobal < queueAddressManager.getCheckpointSuccessStartAddr()) {
+            throw new TrimmedException();
+        }
+
+        // We always have to fill to the read queue to ensure we read up to
+        // max global.
+        if (readQueueIsEmpty) {
+            return Collections.emptyList();
+        }
+
+        List<Long> readList = queueAddressManager.getAllReads(maxGlobal);
+
+        // The list to store read results in
+        List<ILogData> readResults = readAll(readList).stream()
+                .filter(x -> x.getType() == DataType.DATA)
+                .filter(x -> x.containsStream(id))
+                .collect(Collectors.toList());
+
+        // Clear the entries which were read
+        queueAddressManager.clearQueue(maxGlobal);
+
+        // Transfer the addresses of the read entries to the resolved queue
+        readResults.stream()
+                .forEach(x -> queueAddressManager.addToResolvedQueue(x.getGlobalAddress()));
+
+        // Update the global pointer
+        if (!readResults.isEmpty()) {
```

Comment:
Updating global address inside this function seems duplicated. Function remainingUpTo does this job.

Reply:
I'm not sure I understand. Here we update the pointer to the last of read results, while the remainingUpTo uses the same method to update to the maxGlobal given in that api.